experimenter/experimenter/nimbus_ui_new/forms.py

```diff
@@ -35,6 +36,40 @@ def save(self, *args, **kwargs):
         return experiment
 
 
+class NimbusSerializerErrorMixin:
+    def clean(self):
+        if self.show_validation_errors:
+            cleaned_data = super().clean()
+
+            for key, value in cleaned_data.items():
+                if isinstance(value, models.QuerySet):
+                    cleaned_data[key] = [item.pk for item in value]
+
+            for field_name in [
+                "excluded_experiments_branches",
+                "required_experiments_branches",
+            ]:
+                if field_name in cleaned_data:
+                    slugs = cleaned_data[field_name]
+                    pks = []
+                    for slug in slugs:
+                        experiment_slug = slug.split(":")[0]
+                        experiment = NimbusExperiment.objects.get(slug=experiment_slug)
+                        pks.append(experiment.pk)
```

Comment:
This loop is gonna do a bunch of single get queries. We should optimize it to do a single query only returning the primary keys:

```python
slugs = [
    slug_and_branch.split(":")[0]
    for slug_and_branch in cleaned_data[field_name]
]
pks = list(NimbusExperiment.objects.filter(slug__in=slugs).values_list("pk"))
```